### ISSUE TITLE
Error: could not deserialize database, Caused by:     io error: failed to fill whole buffer

### ISSUE THREAD

**Issue by gregnordin** [warpdotdev/Warp#4976]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

After updating to v0.2024.05.28.08.02.stable_00 on Macbook pro M1 Max (Sonoma 14.4.1), I see the following error after every command (all of which seem to execute properly):

```
Error: could not deserialize database

Caused by:
    io error: failed to fill whole buffer
```

### To reproduce

I don't know how to make this reproducible as it occurred with an update of Warp.

### Screenshots

![Screenshot 2024-06-02 at 4 17 53 PM](https://github.com/warpdotdev/Warp/assets/4380622/31a98a6e-272b-4b51-b7b7-c45c8d7aef01)

### Operating system

MacOS

### Operating system and version

Sonoma 14.4.1

### Shell Version

GNU bash, version 5.1.16(1)-release (aarch64-apple-darwin21.1.0)

### Current Warp version

v0.2024.05.28.08.02.stable_00

### Regression

Yes, this bug started recently or with an X Warp version

### Recent working Warp date

A few days ago before updating to latest version of Warp.

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Hey @gregnordin Sorry to hear about this. It's possible that there is an issue with Warps sqlite db, can you run this command `ls $HOME/Library/Application Support/dev.warp.Warp-Stable/` and let us know the result as well as provide the log file is located at `~/Library/Logs/warp.log`?

**Comment by gregnordin** [comment]
Hi Danny,

Here is the result of running the command, and the log file is attached.

Thanks,
Greg

***@***.***



On Jun 3, 2024, at 2:38 PM, Danny Neira ***@***.***> wrote:



Hey @gregnordin<https://github.com/gregnordin> Sorry to hear about this. It's possible that there is an issue with Warps sqlite db, can you run this command ls $HOME/Library/Application Support/dev.warp.Warp-Stable/ and let us know the result as well as provide the log file is located at ~/Library/Logs/warp.log?

**Comment by dannyneira** [comment]
Sorry about this, i didnt escape the space to the command failed. Try running. `ls "$HOME/Library/Application Support/dev.warp.Warp-Stable/"`

You should see a `warp.sqlite` file in there which is the local Warp db for restoration, rich history, etc.

**Comment by gregnordin** [comment]
@dannyneira, here it is:

```
ls "$HOME/Library/Application Support/dev.warp.Warp-Stable/"
65f09a07fbe5953e warp.sqlite      warp.sqlite-shm  warp.sqlite-wal  warp_network.log
```

**Comment by dannyneira** [comment]
It looks like there could be something in your rc file causing this issue and Warp is unable to parse it. 

We have a list of known incompatible tools here: <https://docs.warp.dev/help/known-issues#list-of-incompatible-tools>

You can disable parts of your dotfiles just for Warp by using this conditional statement:

```sh
# Bash and Zsh
if [[ $TERM_PROGRAM != "WarpTerminal" ]]; then
# > What you want to disable here <
fi
```

**Comment by gregnordin** [comment]
I commented out the following line in my `.bash_profile` file and the problem went away.

` eval "$(zoxide init bash)"`

Also, attached is my log file.

**Comment by gregnordin** [comment]
[warp.log](https://github.com/user-attachments/files/15598865/warp.log)

**Comment by dannyneira** [comment]
Thanks for the info. The logs don't indicate any Warp errors that are related. I also tested this and zoxide was working on Warp for me with no errors. It appears that error could be related to the corruption of the Zoxide DB. https://github.com/ajeetdsouza/zoxide/issues/594
On MacOS, the file is typically located at `~/.local/share/zoxide/db.zo`.

Are you seeing the same error on other terminals aswell?

**Comment by gregnordin** [comment]
Yes, I see the same error when I open iTerm2. I found I had an old version of zoxide (0.2) and updated to the latest version. Now there is a different zoxide problem that I need to debug so I think we can close this issue with Warp. Thanks for all of your help!
